### PR TITLE
Add redirect route configuration too allow setting of redirect location when triggered by middleware

### PR DESCRIPTION
**Pull request by joearcher**
I've added a config var for a route name and amended the middlewares to redirect to the configured route, defaulting to login.

### CODE REVIEW

File: src/Http/Middleware/ForbidBannedUser.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function handle($request, Closure $next)
         $user = $this->auth->user();
 
         if ($user && $user instanceof BannableContract && $user->isBanned()) {
-            return redirect()->back()->withInput()->withErrors([
+            return redirect()->route(config('ban.route', 'login'))->withInput()->withErrors([
```
Comment (antonkomarev) [inline src/Http/Middleware/ForbidBannedUser.php:50]: This change might lead to recursion if login page redirects logged in user, then he will be redirected back to login and then redirected forward and then back again.

**Comment by antonkomarev** [comment]
Thank you for the contribution @joearcher!
This is breaking changes and I'm not sure we should release major version with this changes only.
Maybe it would be better to have `null` as default config value and if so - make it work as it was working right now.

**Comment by antonkomarev** [comment]
Another one point that not all people have login route by default and it will crash application. It might be more safe and transparent for the user if it will be redirect url, not route.

Then people might reconfigure it like this if they want to:
```php
'redirect_url' => route('login'),
```

**Comment by joearcher** [comment]
Both of your points make perfect sense, I will amend this to default to a null config value fall back to current behaviour and use a url for the redirect where set.

File: config/ban.php [hunk 2]
```diff
@@ -24,5 +24,17 @@
     */
 
     'load_default_migrations' => true,
-
+   
+    /*
+    |---------------------------------------------------------------------------
+    | RURL to redirect banned user to
```
Comment (antonkomarev) [inline config/ban.php:30]: RURL? :smile:

File: src/Http/Middleware/ForbidBannedUser.php [hunk 3]
```diff
@@ -45,11 +45,20 @@ public function __construct(Guard $auth)
...
-                'login' => 'This account is blocked.',
-            ]);
+            if($redirect_url){
+                return redirect($redirect_url)->withInput()->withErrors([
+                    'login' => 'This account is blocked.',
+                ]);
+            }
+            else{
+                return redirect()->back()->withInput()->withErrors([
+                    'login' => 'This account is blocked.',
```
Comment (antonkomarev) [inline src/Http/Middleware/ForbidBannedUser.php:58]: Errors array is duplicated and could be extracted to the variable.

File: src/Http/Middleware/ForbidBannedUser.php [hunk 4]
```diff
@@ -45,11 +45,20 @@ public function __construct(Guard $auth)
     public function handle($request, Closure $next)
     {
         $user = $this->auth->user();
+        $redirect_url = config('ban.redirect_url', false);
```
Comment (antonkomarev) [inline src/Http/Middleware/ForbidBannedUser.php:48]: `$redirect_url` -> `$redirectUrl`

Comment (antonkomarev) [inline src/Http/Middleware/ForbidBannedUser.php:48]: I'd prefer to have `null` default value and check for `$redirectUrl === null` to determine what case should be used. First of all by this way we will avoid mixed variable types `string|bool` and will use nullable string.

**Review by antonkomarev — CHANGES_REQUESTED** [review]
StyleCI found issues in code style. Fix them to make it green, please.

File: config/ban.php [hunk 5]
```diff
@@ -24,5 +24,17 @@
...
+    |---------------------------------------------------------------------------
+    |
+    | Provide a url, this is where a banned user will be redirected to
```
Comment (antonkomarev) [inline config/ban.php:33]: Some URL example values (absolute, relative, route) in PHPDoc will make it more useful.

File: config/ban.php (hunk 2 above)
Comment (joearcher) [reply config/ban.php]: Styleci now showing green

File: src/Http/Middleware/ForbidBannedUser.php (hunk 3 above)
Comment (joearcher) [reply src/Http/Middleware/ForbidBannedUser.php]: Removed error array duplication

File: src/Http/Middleware/ForbidBannedUser.php (hunk 4 above)
Comment (joearcher) [reply src/Http/Middleware/ForbidBannedUser.php]: Now checking for null

File: config/ban.php (hunk 5 above)
Comment (joearcher) [reply config/ban.php]: Added examples

File: src/Http/Middleware/ForbidBannedUser.php (hunk 4 above)
Comment (joearcher) [reply src/Http/Middleware/ForbidBannedUser.php]: renamed variable to studly case

File: src/Http/Middleware/ForbidBannedUser.php [hunk 6]
```diff
@@ -45,11 +45,17 @@ public function __construct(Guard $auth)
     public function handle($request, Closure $next)
     {
         $user = $this->auth->user();
+        $redirectUrl = config('ban.redirect_url', null);
```
Comment (antonkomarev) [inline src/Http/Middleware/ForbidBannedUser.php:48]: It would be better to move this variables inside of the `if` block, because it's using only there.

Comment (joearcher) [reply src/Http/Middleware/ForbidBannedUser.php]: Moved vars inside if block

**Review by antonkomarev — APPROVED** [review]

**Comment by antonkomarev** [comment]
Thank you for polishing it up! Good job!

**Comment by antonkomarev** [comment]
Will try to make a release today evening.

**Comment by joearcher** [comment]
Happy to help! Thanks for all of your input and patience.

**Comment by antonkomarev** [comment]
[Laravel Ban v4.4.0](https://github.com/cybercog/laravel-ban/releases/tag/4.4.0) has been released.

**Pull request merged**